shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -45,18 +100,185 @@ impl RedisConfig {
 pub struct SimpleRedisCache {
     cache_chain: TransformChain,
     caching_schema: HashMap<String, TableCacheSchema>,
+    missed_requests: Counter,
 }
 
 impl SimpleRedisCache {
     fn get_name(&self) -> &'static str {
         "SimpleRedisCache"
     }
 
-    async fn get_or_update_from_cache(
+    /// Build the messages for the cache query from the cassandra request messages.
+    /// returns the Redis Messages or a `CacheableState:Err` or `CacheableState::Skip` as the error
+    fn build_cache_query(
         &mut self,
-        mut messages_cass_request: Messages,
-    ) -> ChainResponse {
-        // This function is a little hard to follow, so heres an overview.
+        cassandra_messages: &mut Messages,
+    ) -> Result<Messages, CacheableState> {
+        let mut messages_redis_request = Vec::with_capacity(cassandra_messages.len());
+        for cass_request in cassandra_messages {
+            match &mut cass_request.frame() {
```

Comment:
This can be just `match cass_request.frame() {` because `frame()` already returns an `&mut`